File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
from collections import deque
import math

def get_k_num(n, k, lists):
    a = n // k
    b = n % k
    lists.appendleft(str(b))
    if a == 0 :
        return lists
    else :
        return get_k_num(a, k, lists)

def is_prime_number(x):
    i = 2
    while i * i <= x:
        if x % i == 0:
            return False
        i += 1
    return True

def solution(n, k):
    lists = deque()
    get_k_num(n, k, lists)
    
    nums = ''.join(lists).split('0')
    
    result = 0
    for num in nums:
        if num != '' and num != '1':
            if is_prime_number(int(num)):
                # print(num)
                result += 1
    
    return result

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function getKNum(n, k, nums) {
    // 나머지를 push
    nums.push(n % k);

    // 몫이 0이면 종료
    if (Math.floor(n / k) === 0) {
        return nums;
    } else {
        return getKNum(Math.floor(n / k), k, nums);
    }
}

function isPrime(n) {
    if (n < 2) return false;

    let i = 2;
    while (i * i <= n) {
        if (n % i === 0) return false;
        i++;
    }
    return true;
}

function solution(n, k) {
    var answer = 0;
    var lists = [];
    var primes = [];
    
    getKNum(n, k, lists);
    
    var sortLists = lists.reverse().join('').split(0);
    
    for (let i=0; i < sortLists.length; i++) {
        if (isPrime(sortLists[i]) == true) {
            answer += 1;
        }
    }
    
    return answer;
}